Sources/OpenTelemetrySdk/Trace/SpanProcessors/BatchSpanProcessor.swift

```diff
@@ -148,11 +236,39 @@ private class BatchWorker: Thread {
     timeoutTimer.cancel()
   }
   
-  private func exportAction(spanList: [ReadableSpan], explicitTimeout: TimeInterval? = nil) {
-    stride(from: 0, to: spanList.endIndex, by: maxExportBatchSize).forEach {
-      var spansToExport = spanList[$0 ..< min($0 + maxExportBatchSize, spanList.count)].map { $0.toSpanData() }
-      willExportCallback?(&spansToExport)
-      spanExporter.export(spans: spansToExport, explicitTimeout: explicitTimeout)
+    private func exportAction(spanList: [ReadableSpan], explicitTimeout: TimeInterval? = nil) {
+        stride(from: 0, to: spanList.endIndex, by: maxExportBatchSize).forEach {
+            var spansToExport = spanList[$0 ..< min($0 + maxExportBatchSize, spanList.count)].map { $0.toSpanData() }
+            willExportCallback?(&spansToExport)
+            let result = spanExporter.export(spans: spansToExport, explicitTimeout: explicitTimeout)
+            if result == .success {
+                cond.lock()
+                processedSpansCounter?.add(value: spanList.count, attribute: exportedAttrs)
+                cond.unlock()
+            }
+        }
+    }
+}
+
+public enum GuageError: Error {
```

Comment:
If we are not going to handle the error it should use as? LongGaugeBuilderSdk in the usage area instead of creating new extensions. This pattern is not used in any other area